Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm

```diff
@@ -1470,8 +1470,11 @@ static RecommendDesktopClassBrowsingForRequest desktopClassBrowsingRecommendedFo
     if (Quirks::needsIPadMiniUserAgent(request.url()))
         return RecommendDesktopClassBrowsingForRequest::No;
 
-    if (Quirks::needsIPhoneUserAgent(request.url()))
-        return RecommendDesktopClassBrowsingForRequest::No;
+    if (Quirks::needsIPhoneUserAgent(API::WebsitePolicies& policies, request.url())) {
+        policies.setCustomUserAgent(makeStringByReplacingAll(standardUserAgentWithApplicationName(m_applicationNameForUserAgent), "iPad"_s, "iPhone"_s));
+        policies.setCustomNavigatorPlatform("iPhone"_s);
```

Comment:
You use the `policies` object here, which is fine since it is in the UIProcess (at the WebKit level). No need to pass it to the `WebCore::Quirks` layer as far as I can tell.

Reply:
yes, I overthought it. 
```
const API::WebsitePolicies& policies = m_configuration->defaultWebsitePolicies();
```